python/py-python-socks/Portfile

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-python-socks
+version             2.0.3
+revision            0
+license             Apache2
+maintainers         nomaintainer
+
+description         Core proxy (SOCKS4, SOCKS5, HTTP tunneling) functionality
+long_description    The python-socks package provides a core proxy \
+                    client functionality for Python. Supports \
+                    SOCKS4(a), SOCKS5(h), HTTP (tunneling) proxy and \
+                    provides sync and async (asyncio, trio, curio, \
+                    anyio) APIs. You probably don't need to use \
+                    python-socks directly. It is used internally by \
+                    aiohttp-socks and httpx-socks packages.
+
+homepage            https://github.com/romis2012/python-socks
+
+python.versions     38 39 310
+
+checksums           rmd160  2e6f958157a28322c995013200d7ad4f31e8b3a9 \
+                    sha256  e3a9ca8e554733862ce4d8ce1d10efb480fd3a3acdafd03393943ec00c98ba8a \
+                    size    24877
+
+if {${name} ne ${subport}} {
+    python.pep517   yes
+
+    depends_run-append  \
+                    port:py${python.version}-anyio \
+                    port:py${python.version}-async-timeout \
+                    port:py${python.version}-attrs \
+                    port:py${python.version}-curio \
+                    port:py${python.version}-flask \
+                    port:py${python.version}-psutil \
+                    port:py${python.version}-trio \
+                    port:py${python.version}-yarl
+}
```

Comment:
there are only four optional dependencies listed in the [README](https://github.com/romis2012/python-socks/tree/v2.0.3#requirements) and in [`setup.py`](https://github.com/romis2012/python-socks/blob/c43f061905f1d7384127efce4708c8da26ef63cc/setup.py#L54) - where do all the others come from?

Reply:
Fixed thanks after double-checking with searches, e.g. https://github.com/romis2012/python-socks/search?q=yarl.

Reply:
I'm not sure whether I agree that this is fixed. If you look that the search you mention in your comment above you'll notice that it is only imported in tests. So why is it runtime dependency and not a test dependency? Again, only the four dependencies that are listed as optional as I stated in my first comment, are used in the actual code. All others seem to be only used in tests and I would assume are, therefore, only test dependencies - or?

Reply:
Okay now I think I have it right. Thank you very much for pointing me in the right direction.